pool/pool.go

```diff
@@ -53,7 +53,8 @@ func (p *Pool) GetN(N int) *blockchain.Transaction {
 
 // GetIndex returns the index of the transaction in the ordering.
 func (p *Pool) GetIndex(t *blockchain.Transaction) int {
```

Comment:
We lose the property that attempting to add a transaction with an input that already exists will fail. Now we can do [IN1, IN2]->OUT1 and [IN1, IN3]->OUT2 and addition will succeed. What are we doing to mediate this and ensure we don't add to the pool if an input already exists? A many-to-one mapping for inputs -> transactions would be nice. Like IN1->OUT1, IN2->OUT2.

Reply:
Its actually ok that multiple transactions in the pool exist with the same input. A single transaction may have awarded 2 coins to alice and 1 to bob, that transaction will serve as an input in at most 2 transactions in the future.

I'll admit that this implementation isn't quite perfect though 😬.

Reply:
The same situation goes for testing double spends within blocks. Its ok that an input is referenced more than once, but not if its references more than once by the same sender. Thanks to your map idea, we now do that check in linear time though :tada: